hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+
+        # disactivate line
+        self.free_background()
+        with stash_active_state(self):
+            self.disable_xray_lines()
+            self.set_signal_range(start_energy, end_energy)
+            for component in self:
+                if component.isbackground is False:
+                    self.remove_signal_range(
+                        component.centre.value -
+                        windows_sigma[0] * component.sigma.value,
+                        component.centre.value +
+                        windows_sigma[1] * component.sigma.value)
+            if kind == 'single':
+                self.fit(**kwargs)
+            if kind == 'multi':
+                self.multifit(**kwargs)
+            self.reset_signal_range()
+        self.fix_background()
+
+    def _twin_xray_lines_width(self, xray_lines):
+        """
+        Twin the width of the peaks
+
+        The twinning models the energy resolution of the detector
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            if i == 0:
+                component_ref = component
+                component_ref.sigma.free = True
+                E_ref = component_ref.centre.value
+            else:
+                component.sigma.free = True
+                E = component.centre.value
+                component.sigma.twin_function = _get_sigma(
+                    E, E_ref, self.units_factor)
+                component.sigma.twin_inverse_function = _get_sigma(
+                    E_ref, E, self.units_factor)
+                component.sigma.twin = component_ref.sigma
+
+    def _set_energy_resolution(self, xray_lines, *args, **kwargs):
+        """
+        Adjust the width of all lines and set the fitted energy resolution
+        to the spectrum
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        energy_Mn_Ka, FWHM_MnKa_old = self.spectrum._get_line_energy('Mn_Ka',
+                                                                     'auto')
+        FWHM_MnKa_old *= eV2keV / self.units_factor
+        get_sigma_Mn_Ka = _get_sigma(
+            energy_Mn_Ka, self[xray_lines[0]].centre.value, self.units_factor)
+        FWHM_MnKa = get_sigma_Mn_Ka(self[xray_lines[0]].sigma.value
+                                    ) * eV2keV / self.units_factor * sigma2fwhm
+        if FWHM_MnKa < 110:
+            warnings.warn("FWHM_MnKa of " + str(FWHM_MnKa) + " smaller than" +
```

Comment:
I would use `ValueError` instead of a warning. The execution should stop here any way